src/hooks/usePagination.tsx

```diff
@@ -0,0 +1,48 @@
+export const usePagination = (
+  currentPage: number,
+  postsTotal: number | undefined
```

Comment:
Pagination should not know about posts, so `postsTotal` => `totalItems`.

Also, `totalItems` should never be `undefined`